tutorials/incorporating_oli_calculations.ipynb

```diff
@@ -234,43 +296,85 @@
+    "        # save chemistry information\n",
+    "        chemistry_info = file_summary[\"chemistry_info\"]\n",
+    "        write_output(chemistry_info[\"result\"], \"chemistry_info\")\n",
+    "        \n",
+    "        # create water analysis input\n",
+    "        # Water Analysis uses true species for inputs, i.e., IONS.\n",
+    "        json_input = flash.configure_water_analysis(\n",
+    "            inflows,\n",
+    "            reconciliation=\"ReconcilePh\",\n",
+    "            ph=5,\n",
+    "            allow_solids=True,\n",
+    "            file_name=\"water_analysis_inputs\",\n",
     "        )\n",
-    "    \n",
-    "        # create water analysis case\n",
-    "        water_analysis_base_case = f.run_flash(\n",
+    "        # run Water Analysis flash calculation survey as specified        \n",
+    "        stream_output = flash.run_flash(\n",
     "            \"wateranalysis\",\n",
     "            oliapi,\n",
     "            dbs_file_id,\n",
-    "            water_analysis_input,\n",
-    "            file_name=\"water_analysis_singlepoint\"\n",
+    "            json_input,\n",
+    "            survey,\n",
+    "            file_name=\"water_analysis_outputs\",\n",
     "        )\n",
     "        \n",
-    "        # create apparent composition from water analysis output\n",
-    "        water_analysis_apparent_composition = f.build_flash_calculation_input(\n",
-    "            \"isothermal\",\n",
-    "            state_vars,\n",
-    "            water_analysis_base_case[0],\n",
+    "        # The output of Water Analysis gives apparent species\n",
+    "        # i.e., SALTS, COMPLEXES (MgO, CaO, etc.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if credential_manager:\n",
+    "    with OLIApi(credential_manager) as oliapi:\n",
+    "\n",
+    "        # Other flash calculations use apparent species for inputs.\n",
+    "\n",
+    "        # These can be obtained in one step with get_apparent_species_from_true(),\n",
+    "        # and saved to a file and reloaded as needed.\n",
+    "        apparent_species = flash.get_apparent_species_from_true(\n",
+    "            json_input, oliapi, dbs_file_id, file_name=\"apparent_species\"\n",
     "        )\n",
-    "        \n",
-    "        # run isothermal flash for the survey parameters\n",
-    "        isothermal_survey_result = f.run_flash(\n",
+    "        isothermal_input = flash.configure_flash_analysis(\n",
+    "            apparent_species, \"isothermal\", file_name=\"isothermal_inputs\"\n",
+    "        )\n",
+    "        # specify a new survey to compare apparent species\n",
+    "        isothermal_survey = build_survey(\n",
+    "            {\"NaCl\": linspace(0, 1e6, 10)},\n",
+    "            get_oli_names=True,\n",
+    "            file_name=\"isothermal_survey\",\n",
+    "        )\n",
+    "        flash.run_flash(\n",
```

Comment:
Running this `run_flash` call will result in an error. I also came across this area in my own script when trying to run this functionality. Here's the traceback:
```
TypeError                                 Traceback (most recent call last)
Cell In[11], [line 20](vscode-notebook-cell:?execution_count=11&line=20)
     [14](vscode-notebook-cell:?execution_count=11&line=14) # specify a new survey to compare apparent species
     [15](vscode-notebook-cell:?execution_count=11&line=15) isothermal_survey = build_survey(
     [16](vscode-notebook-cell:?execution_count=11&line=16)     {"NaCl": linspace(0, 1e6, 10)},
     [17](vscode-notebook-cell:?execution_count=11&line=17)     get_oli_names=True,
     [18](vscode-notebook-cell:?execution_count=11&line=18)     file_name="isothermal_survey",
     [19](vscode-notebook-cell:?execution_count=11&line=19) )
---> [20](vscode-notebook-cell:?execution_count=11&line=20) flash.run_flash(
     [21](vscode-notebook-cell:?execution_count=11&line=21)     "isothermal",
     [22](vscode-notebook-cell:?execution_count=11&line=22)     oliapi,
     [23](vscode-notebook-cell:?execution_count=11&line=23)     dbs_file_id,
     [24](vscode-notebook-cell:?execution_count=11&line=24)     isothermal_input,
     [25](vscode-notebook-cell:?execution_count=11&line=25)     isothermal_survey,
     [26](vscode-notebook-cell:?execution_count=11&line=26)     file_name="isothermal_outputs",
     [27](vscode-notebook-cell:?execution_count=11&line=27) )

File [c:\users\adam\my-nawi-hub\watertap\watertap\tools\oli_api\flash.py:889](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:889), in Flash.run_flash(self, flash_method, oliapi_instance, dbs_file_id, json_input, survey, file_name)
    [883](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:883) for idx in range(num_samples):
    [884](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:884)     _logger.info(f"Flash sample #{idx+1} of {num_samples}")
    [885](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:885)     requests_to_process.append(
    [886](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:886)         {
    [887](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:887)             "flash_method": flash_method,
    [888](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:888)             "dbs_file_id": dbs_file_id,
--> [889](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:889)             "input_params": self.get_clone(
    [890](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:890)                 flash_method, json_input, idx, survey
    [891](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:891)             ),
    [892](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:892)         }
    [893](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:893)     )
    [894](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:894) processed_requests = oliapi_instance.process_request_list(
    [895](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:895)     requests_to_process,
    [896](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:896)     # burst_job_tag=burst_job_tag,
    [897](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:897)     # max_concurrent_processes=max_concurrent_processes,
    [898](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:898)     # batch_size=batch_size,
    [899](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:899) )
    [900](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:900) _logger.info("Completed running flash calculations")

File [c:\users\adam\my-nawi-hub\watertap\watertap\tools\oli_api\flash.py:963](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:963), in Flash.get_clone(self, flash_method, json_input, index, survey)
    [961](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:961)     _logger.warning(f"Survey key {k} not found in JSON input.")
    [962](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:962) if self.relative_inflows:
--> [963](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:963)     if "value" in d[k]:
    [964](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:964)         d[k]["value"] += v[index]
    [965](file:///C:/users/adam/my-nawi-hub/watertap/watertap/tools/oli_api/flash.py:965)     else:

TypeError: argument of type 'float' is not iterable

```